apps/snitch_api/lib/snitch_api_web/views/product_view.ex

```diff
@@ -37,6 +41,17 @@ defmodule SnitchApiWeb.ProductView do
     product.selling_price |> to_string
   end
 
+  def default_image(product, _conn) do
+    default_image = from(image in Image, where: image.is_default == true)
+
+    query =
+      from(p in ProductSchema, where: p.id == ^product.id, preload: [images: ^default_image])
+
+    product = Repo.one(query)
```

Comment:
Handle clause where we are not able to fetch the default image.

Steps to reproduce:
1. Add images to any `active` product. 
2. Delete the default image.
3. Try to call the product listing API.

API will fail.